bin/backup/start-backupjob.sh

```diff
@@ -40,7 +40,11 @@ env_check_info "BACKUP_LABEL" "BACKUP_LABEL is set to ${BACKUP_LABEL}."
 env_check_info "BACKUP_PATH" "BACKUP_PATH is set to ${BACKUP_PATH}."
 env_check_info "BACKUP_OPTS" "BACKUP_OPTS is set to ${BACKUP_OPTS}."
 
-echo "*:*:*:"${BACKUP_USER?}":"${BACKUP_PASS?} >> ${PGPASSFILE?}
+# escape any instances of ':' or '\' with '\' in the provided password
+# before storing the value in the password file
+ESCAPED_PASSWORD=$(echo ${BACKUP_PASS?} | sed 's/[:\\]/\\&/g')
```

Comment:
I like the simplicity of this `sed` expression. We can eliminate the `echo` using [Bash here string](https://wiki.bash-hackers.org/syntax/redirection#here_strings). With this, the assignment will have the same exit code as `sed`.

```suggestion
ESCAPED_PASSWORD=$(sed <<< "${BACKUP_PASS?}" 's/[:\\]/\\&/g')
```

---

If we _don't_ want to use `sed`, I think the same effect can be had with Bash parameter expansion. It's easier to get wrong (backslash needs to be replaced before colon; don't accidentally swap them) but it does save one process spawn. I prefer `sed`, I think.

```suggestion
ESCAPED_PASSWORD="${BACKUP_PASS?}"
ESCAPED_PASSWORD="${ESCAPED_PASSWORD//\\/\\\\}"
ESCAPED_PASSWORD="${ESCAPED_PASSWORD//:/\\:}"
```

Reply:
I like the here string change, I'll implement that. I think, with sed being a bit easier to 'get right', I'll also agree to stick with that. Thanks!